### ISSUE TITLE
Accessible dropdown button (for audio downloads)

### ISSUE THREAD

**Issue by zackkrida** [WordPress/openverse-frontend#149]

**Comment by panchovm** [comment]
I was thinking of a dropdown menu similar to the _filters applied_ menu when hovering the filter badge next to expanding filters action. Yet a pending mockup I need to polish.

**Comment by sarayourfriend** [comment]
Is there a figma for the dropdown yet?

Also, when clicking one of the dropdown options, does it immediately trigger the download or does it just change the format that will be downloaded when clicking download?

**Comment by zackkrida** [comment]
> Is there a figma for the dropdown yet?

> Also, when clicking one of the dropdown options, does it immediately trigger the download or does it just change the format that will be downloaded when clicking download?

One more question from me:

> Anything on the button to show what the currently-selected format is? Like "download audio" is essentially the button label, can we add some text for "MP3 V0" or whatever the selected format is?

These are good questions for @panchovm 😄 . I think the answers are both "not yet" and "but soon" for all questions.

**Comment by panchovm** [comment]
Haha, like @zackkrida said, not yet but soon. Although I have to say that I was thinking of **immediately trigger the download**. However, in all audio views, the default option will be the same and the user may have to select the favorite option every time, which might be annoying in the long-term run. Thoughts about that?

Regarding: 

> can we add some text for "MP3 V0" or whatever the selected format is?

Yes. Very considered

**Comment by sarayourfriend** [comment]
> However, in all audio views, the default option will be the same and the user may have to select the favorite option every time, which might be annoying in the long-term run. Thoughts about that?

In implementing the dropdown we could also store their preference in `localStorage` so that their previously selected format is pre-selected the next time they visit an audio result page.

**Comment by zackkrida** [comment]
> In implementing the dropdown we could also store their preference in `localStorage` so that their previously selected format is pre-selected the next time they visit an audio result page.

Which we already do for filter sidebar visibility!


Would the idea be that a click of a drop down choice 1. Downloads the file and 2. Sets the default download format to the clicked format?

**Comment by panchovm** [comment]
That sounds excellent. I have an idea but I need to make it.

**Comment by sarayourfriend** [comment]
> Would the idea be that a click of a drop down choice 1. Downloads the file and 2. Sets the default download format to the clicked format?

What's what I was thinking, if clicking a dropdown choice does indeed trigger a download. If not then just step 2.

**Comment by sarayourfriend** [comment]
@panchovm Have you had a chance to look at these designs yet? I'm at a stopping point where I need the designs to be able to move forward much more 🙂

**Comment by panchovm** [comment]
Yes. I'm working this week on this

**Comment by panchovm** [comment]
https://user-images.githubusercontent.com/895819/131739563-4aa0986b-a00a-4232-947e-be1420ee158b.mp4

This is the idea I like most. The dropdown does not use much space, and the popover style matches the WP Admin hovering styles. I didn't like having the format and file size as labels, but then I noticed the file information is more explicit.

Here are the other ideas rejected.

<img width="500" alt="Screen Shot 2021-09-01 at 16 24 57" src="https://user-images.githubusercontent.com/895819/131739954-a31b1d86-c14e-49ae-a05b-82cd523a01a5.png">

**Comment by sarayourfriend** [comment]
Awesome. The option you chose is easier to program as well. Thanks for the mockup!

**Comment by sarayourfriend** [comment]
@zackkrida should we use the `Intl.NumberFormat` for that?

**Comment by panchovm** [comment]
Here is [the mockup](https://www.figma.com/file/vEoItpMQyFZb5AZV9aoP7I/Audio-integration?node-id=1122%3A13994) where the popover is visible and the first item in hover state. And [the prototype](https://www.figma.com/proto/vEoItpMQyFZb5AZV9aoP7I/Audio-integration?page-id=634%3A6338&node-id=1083%3A12397&viewport=326%2C48%2C0.51&scaling=min-zoom&starting-point-node-id=1083%3A12397) to see all interactions.

**Comment by sarayourfriend** [comment]
@panchovm I'm not sure how to check in Figma, but does this text pass accessible contrast requirements?

<img width="98" alt="Captura de Tela 2021-09-02 às 13 35 43" src="https://user-images.githubusercontent.com/24264157/131912518-b9b4aba3-a104-4e4e-88b6-008628fab2ff.png">

**Comment by zackkrida** [comment]
<img width="255" alt="CleanShot 2021-09-02 at 17 44 31@2x" src="https://user-images.githubusercontent.com/6351754/131920368-9bd05e3e-00ce-4c3d-a247-397c6e6a3593.png">

nope. 
i checked the hex values in firefox devtools.

**Comment by panchovm** [comment]
You are right @sarayourfriend, my mistake. It doesn't meet the AA level. Let's keep it wite. The mockup was updated with the change.
